crates/e2e/tests/e2e/limit_orders.rs

```diff
@@ -0,0 +1,257 @@
+}
+
+async fn single_limit_order_test(web3: Web3) {
+    shared::tracing::initialize_for_tests("warn,orderbook=debug,solver=debug,autopilot=debug");
+    shared::exit_process_on_panic::set_panic_hook();
+    let contracts = crate::deploy::deploy(&web3).await.expect("deploy");
+
+    let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+    let solver_account = Account::Local(accounts[0], None);
+    let trader_account = Account::Offline(PrivateKey::from_raw(TRADER_PK).unwrap(), None);
+
+    // Create & Mint tokens to trade
+    let token_a = deploy_mintable_token(&web3).await;
+    let token_b = deploy_mintable_token(&web3).await;
+
+    // Fund trader and settlement accounts
+    tx!(
+        solver_account,
+        token_a.mint(trader_account.address(), to_wei(100))
+    );
+    tx!(
+        solver_account,
+        token_b.mint(contracts.gp_settlement.address(), to_wei(100))
+    );
+
+    // Create and fund Uniswap pool
+    tx!(
+        solver_account,
+        contracts
+            .uniswap_factory
+            .create_pair(token_a.address(), token_b.address())
+    );
+    tx!(
+        solver_account,
+        token_a.mint(solver_account.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_b.mint(solver_account.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_a.approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_b.approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        contracts.uniswap_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(100_000),
+            to_wei(100_000),
+            0_u64.into(),
+            0_u64.into(),
+            solver_account.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Create and fund pools for fee connections.
+    for token in [&token_a, &token_b] {
+        tx!(
+            solver_account,
+            token.mint(solver_account.address(), to_wei(100_000))
+        );
+        tx!(
+            solver_account,
+            token.approve(contracts.uniswap_router.address(), to_wei(100_000))
+        );
+        tx_value!(solver_account, to_wei(100_000), contracts.weth.deposit());
+        tx!(
+            solver_account,
+            contracts
+                .weth
+                .approve(contracts.uniswap_router.address(), to_wei(100_000))
+        );
+        tx!(
+            solver_account,
+            contracts.uniswap_router.add_liquidity(
+                token.address(),
+                contracts.weth.address(),
+                to_wei(100_000),
+                to_wei(100_000),
+                0_u64.into(),
+                0_u64.into(),
+                solver_account.address(),
+                U256::max_value(),
+            )
+        );
+    }
```

Comment:
As a general note (and unrelated to this PR), do you think it makes sense to have this stuff be initialized as part of a test fixture? Might make adding new E2E tests easier in the future.